services/src/test/java/org/apache/druid/testing/simulate/EmbeddedDruidServer.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.testing.simulate;
+
+import com.google.inject.Binder;
+import com.google.inject.Injector;
+import org.apache.druid.cli.ServerRunnable;
+import org.apache.druid.client.broker.BrokerClient;
+import org.apache.druid.client.coordinator.CoordinatorClient;
+import org.apache.druid.indexing.overlord.IndexerMetadataStorageCoordinator;
+import org.apache.druid.java.util.common.StringUtils;
+import org.apache.druid.java.util.common.lifecycle.Lifecycle;
+import org.apache.druid.java.util.common.logger.Logger;
+import org.apache.druid.query.DruidProcessingConfigTest;
+import org.apache.druid.rpc.indexing.OverlordClient;
+import org.apache.druid.server.DruidNode;
+import org.apache.druid.server.metrics.LatchableEmitter;
+import org.apache.druid.utils.RuntimeInfo;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * An embedded Druid server used in simulation tests.
+ * This class and most of its methods are kept package protected as they are used
+ * only by the specific server implementations in the same package.
+ */
+public abstract class EmbeddedDruidServer implements ServerReferencesProvider
```

Comment:
since almost every class is called `EmbeddedX` ; I feel like it would  probably be an option to call these "embedded tests" ?

Reply:
my 2¢: to me the word "simulation" is misleading given the final design- I don't believe anything is really being simulated. IMO using "embedded" across the board is better. I suggest searching for "simulate" or "simulation" and adjusting them so we can keep the wording consistent. As for `XyzSimTest`, could be `XyzEmbedTest`.

Reply:
Sounds good, will make the change.